### PR TITLE
Mod C

### PR DESCRIPTION
**Pull request by folkertdev**

### CODE REVIEW

File: slides/C-concurrency-parallelism.md [hunk 1]
```diff
@@ -0,0 +1,613 @@
...
+info: "Rust - X: Y"
+drawings:
+  persist: false
+fonts:
+  mono: Fira Mono
+layout: cover
+title: 'Rust - X: Y'
+---
+# Rust programming
+Module X: description
```
Comment (squell) [inline slides/C-concurrency-parallelism.md:15]: Module C?

File: slides/C-concurrency-parallelism.md [hunk 2]
```diff
@@ -0,0 +1,613 @@
...
+<!-- Introduce today's subject -->
+
+
+---
+layout: default
+---
+# Learning objectives
+<!-- List this module's learning objectives -->
+
+- Working with C from Rust and vice versa
```
Comment (squell) [inline slides/C-concurrency-parallelism.md:73]: This also needs to be updated?

File: slides/C-concurrency-parallelism.md [hunk 3]
```diff
@@ -0,0 +1,613 @@
...
+```rust
+let numbers = Vec::from_iter(0..=1000);
+
+let t = thread::spawn(|| {
+    let len = numbers.len();
+    let sum = numbers.iter().sum::<usize>();
+    sum / len
+});
+
+drop(numbers); // oh no
```
Comment (squell) [inline slides/C-concurrency-parallelism.md:372]: Doesn't the Rust type system slap you on the wrist here? The way the slide is made with the "oh no") makes it look like you are on your own, as you would in C.

File: slides/C-concurrency-parallelism.md [hunk 4]
```diff
@@ -0,0 +1,613 @@
...
+        let len = numbers.len();
+        let sum = numbers.iter().sum::<usize>();
+        sum / len
+    }).join().unwrap()
+});
+
+println!("average: {average:?}");
+```
+
+- explicitly bound the lifetime with a scope
```
Comment (squell) [inline slides/C-concurrency-parallelism.md:399]: I'm a bit confused by calling this 'explicit' as the thread::scope machinery actually makes it 'implicitly' safe to borrow non-'static things. "Explicit" to me would mean I would expect to see a `'a` somewhere, or some such.

**Review by squell — APPROVED** [review]

File: slides/C-concurrency-parallelism.md [hunk 5]
```diff
@@ -0,0 +1,606 @@
...
+---
+## Who am i?
+::left::
+- Ferris
+- I Love Rust
+
+::right::
+<img src="https://rustacean.net/assets/rustacean-orig-noshadow.png" alt="Photo Ferris" width="300" />
+<!-- Optionally quickly introduce yourself, add photo -->
+
```
Comment (hdoordt) [inline slides/C-concurrency-parallelism.md:30]: You can remove stuff from the template you are note going to use

File: slides/C-concurrency-parallelism.md [hunk 6]
```diff
@@ -0,0 +1,606 @@
...
+---
+layout: default
+---
+# Last time...
+- Ownership model
+- Move semantics
```
Comment (hdoordt) [inline slides/C-concurrency-parallelism.md:36]:
```suggestion
- Cargo and dependencies
- Creating a nice API
- Testing and benchmarking
- Setting up your own project
```

File: slides/C-concurrency-parallelism.md [hunk 7]
```diff
@@ -0,0 +1,606 @@
...
+    // evaluate the closure
+    f(1, 2)
+}
+```
+
+- very useful when working with iterators, `Option` and `Result`.
+
+```rust
+let evens: Vec<_> = some_iterator.filter(|x| x % 2 == 0).collect();
+```
```
Comment (hdoordt) [inline slides/C-concurrency-parallelism.md:68]: Maybe do this after the learning objectives

**Review by hdoordt — COMMENTED** [review]
Just some minor things left, almost good to go!

**Review by hdoordt — CHANGES_REQUESTED** [review]
Please add a (small) section in the mdbook as students will expect to find exercise instructions there. It could be as easy as this: <https://101-rs.tweede.golf/A2-advanced-intro/mod.html>. Please annotate the exercise instructions with a number of ⭐s indicating the difficulty, where ⭐ is beginner level and ⭐⭐⭐⭐ is advanced

(Apologies for the emotional roller coaster)

File: slides/C-concurrency-parallelism.md (hunk 7 above)
Comment (folkertdev) [reply slides/C-concurrency-parallelism.md:70]: yeah well this does not fit in anywhere... I'll play around with it

**Review by hdoordt — APPROVED** [review]
Looks good!

**Pull request merged**